### DISCUSSION TITLE
Network View: Can a router be displayed as one device?

### DISCUSSION THREAD

**Discussion by jstars-gmti** [netalertx/NetAlertX#902] [Q&A] [answered]
I am just wondering if I have missed something. My ISP router shows as two devices ("Internet" with public IP address) and another device (call it a gateway) with the LAN IP address. 
<img width="773" alt="Screenshot 2024-12-01 at 13 46 40" src="https://github.com/user-attachments/assets/dcb8c9b0-160b-4896-a7c3-5bd8c251670e">
Is there a way to depict this as one device in the network view, that is, a router with two interfaces?

Another question: I am interested to how are people depicting ethernet jacks, in particular those that have nothing plugged in. They lead to a port on a switch and are available but not active. I'd like to show them as available but not active until something is plugged in to them. Undiscoverable device?

**Comment by jokob-sk** [comment, marked as answer]
Hi @jstars-gmti ,

Your first question might be related to this open feature request: https://github.com/jokob-sk/NetAlertX/issues/724 I'm still not sur ehow to approach this because of the complexity of handling notifications, listings, sessions, and even the network view. 

The second, displaying available ports, is not something I've thought about yet. Knowing how the app is build it wouldn't be too difficult to implement, but the validation if a port is used or not when editing a device could be tricky. I will probably say it's a nice to have feature at this point - if I work on something similar, I will think of this feature request as well. 

Maybe someone form the community will give you ideas or workaround suggestions. The next version will also come with the option to create and edit all aspects of a dummy device, which _might_ help.

**Reply by jstars-gmti** [reply]
Ok thanks. I just wanted to make sure I wasn't missing functionality that I couldn't find. I think the first issue is topical because so many devices have multiple interfaces (wifi, ethernet). The real question is how do you want to depict this: logically, physically, or toggle between the two? I think a physical view would take up less screen space to depict in network view. It really depends on what you want the core functionality of the project to be. At the very least perhaps the two interfaces can be stacked on top of each other (to take less horizontal space) as in the case of a router, these are two different networks.

Regarding the second, this is less essential. But the logic would be that the (undiscoverable) RJ-45 jack would be green if the devices connected to it are green/online, else grey/offline. I have found it odd that my (undiscoverable) unmanned switch shows grey/offline when the devices connected downstream to it are green/online, so this I would think would be easy to fix and implement.

**Reply by jokob-sk** [reply]
Thanks for the ideas! 

Toggling between the two display modes - sounds like a code maintenance nightmare 😅especially if this toggle would also influence notifications. I still am unsure what to do with #724 - not sure if I'll ever come up with something clever for that one. Feel free to contribute to that issues to prompt other peoples' ideas. I'm a bit burned out from the recent refactoring so I'm taking it a bit slowly, I still need to put in some solid effort to refactor the data retrieval on the front end. 

I'll consider this - the idea originally was red - device offline - grey - online (I don't use green in the network display currently - only on the dots on the tabs). I also want to refactor the network display code so I might add additional features when I do that in future (e.g. use datatables instead of a generated table).

As always, thanks for any input and help!
j

**Reply by jstars-gmti** [reply]
Understood. I think whatever solution is found, consistency with colors across the application will be important. As I think about it, to me green intuitively means "go/running/connected/online". Grey to me means "inactive/not-available/offline". Red is usually associated with "no-go/problems/criticality/halts". I like the way the device status is displayed in the My Devices list:  Green is shown as on-line and grey is offline. I'd recommend the same in the network view as well (including the dots). I think that red should be reserved for attention grabbing problems. Again this is not core functionality stuff, only for down the road.
